test/tests/functional/pbs_qsub_direct_write.py

```diff
@@ -88,11 +88,12 @@ def test_direct_write_when_job_succeeds_controlled(self):
                 (but is a gid that the user is a member of)
         3) not accessible via other permissions
         """
-        j = Job(TEST_USER4, attrs={ATTR_k: 'doe'})
+        j = Job(TEST_USER2, attrs={ATTR_k: 'doe'})
         j.set_sleep_time(10)
         sub_dir = self.du.create_temp_dir(asuser=TEST_USER5)
         mapping_dir = self.du.create_temp_dir(
-            asuser=TEST_USER4, asgroup=TSTGRP5, mode=0o770)
+            asuser=TEST_USER2, asgroup=TSTGRP5)
+        self.du.chmod(path=mapping_dir, mode=0o770, runas=TEST_USER2)
```

Comment:
I'm trying to understand the change.

why the user has changed from TEST_USER4 to TEST_USER2?
also, why to use chmod when create_temp_dir accepts mode?

Reply:
I made the change to username because in the test at line 104/105 the test tries to access the temporary directory to list the files. Now unless the user who is running the test and the user who is running the job is part of a common group, it will fail.
pbsroot is part of tstgrp2 and so is user2 and that is why this change.

About chmod, I think you are right. I'm trying another change to ensure that create_temp_dir handles that change.